File: backend/src/main/java/ru/panteleevya/backend/person/Person.java
package ru.panteleevya.backend.person;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Person {
    private String personId;
    private String surname;
    private String name;
    private String nickname;
    private String bio;
    private List<Alias> aliases;
    private List<PersonalChatInfo> personalChats;
    private List<GroupChatInfo> groupChats;
}

File: frontend/src/features/interfaces/person.ts
export interface Alias {
    aliasedPersonId: string;
    aliasedSurname: string;
    aliasedName: string;
}

export interface PersonalChatInfo {
    personId: string;
    chatId: string;
}

export interface GroupChatInfo {
    chatId: string;
}

export interface Person {
    personId: string;
    surname: string;
    name: string;
    nickname: string;
    bio: string;
    aliases: Alias[];
    personalChats: PersonalChatInfo[];
    groupChats: GroupChatInfo[];
}

export const nullPerson: Person = {
    personId: "",
    surname: "",
    name: "",
    nickname: "",
    bio: "",
    aliases: [],
    personalChats: [],
    groupChats: [],
}